File: typescript/strategy.ts
import * as fs from 'fs';

class Context {
    private strategy: Strategy;
    private filename: string;

    constructor(strategy: Strategy, filename: string) {
        this.strategy = strategy;
        this.filename = filename
    }

    private getLimitedData(): string[] {
        let file = fs.readFileSync(this.filename, 'utf-8');
        let file_rows = file.split("\n");
        return file_rows.slice(0, 100);
    }

    public setStrategy(strategy: Strategy) {
        this.strategy = strategy;
    }

    public printData(): void {
        let dataToOutput = this.strategy.getData(this.getLimitedData());
        console.log(dataToOutput);        
    }

}


interface Strategy {
    getData(sourceData: string[]): string[];
}

// first strategy
class DataFilter implements Strategy {
    private moreThan: number;
    
    /* Numbers exceeding the specified value will be added to the array
    default is 10 */
    constructor(moreThan: number = 10) {
        this.moreThan = moreThan
    }

    public getData(sourceData: string[]): string[] {
        let output: string[] = [];
        
        for (let i = 0; i < sourceData.length; i++) {
            if (+sourceData[i] > this.moreThan) {
                output.push(sourceData[i]);
            }
        }
        return output;
    }
}

// second strategy
class DataRaw implements Strategy {
    public getData(sourceData: string[]): string[] {
        return sourceData
    }
}

const context = new Context(new DataFilter(), './input.txt')
console.log('Filter strategy')
context.printData()

context.setStrategy(new DataRaw())
console.log('No filter strategy')
context.printData()

File: python/strategy.py
from abc import ABC, abstractmethod
from typing import List


class Strategy(ABC):
    @abstractmethod
    def get_data(self, source_data: List[int]) -> List[int]:
        return source_data


# first strategy
class DataFilter(Strategy):
    def __init__(self, more_than: int = 10):
        self.__more_than = more_than

    def get_data(self, source_data: List[int]) -> List[int]:
        output = list()
        for line in source_data:
            if line > self.__more_than:
                output.append(line)
        return output


# second strategy
class DataRaw(Strategy):
    def get_data(self, source_data: List[int]) -> List[int]:
        return source_data


class Context:
    def __init__(self, strategy: Strategy, filename: str):
        self.__strategy = strategy
        self.__filename = filename

    def __get_limited_data(self) -> List[int]:
        lines = 100
        with open(self.__filename, 'r') as file:
            file_data = [int(next(file)) for x in range(lines)]
        return file_data

    @property
    def strategy(self):
        return self.__strategy

    @strategy.setter
    def strategy(self, strategy: Strategy):
        self.__strategy = strategy

    def print_data(self):
        return print(self.__strategy.get_data(self.__get_limited_data()))


if __name__ == '__main__':
    context = Context(DataFilter(), 'input.txt')
    print('Filter strategy')
    context.print_data()

    context.strategy = DataRaw()
    print('No filter strategy')
    context.print_data()
